drivers/wifi/esp/esp.h

```diff
@@ -192,6 +198,7 @@ struct esp_data {
 	struct in_addr gw;
 	struct in_addr nm;
 	uint8_t mac_addr[6];
+	struct sockaddr_in dns_addresses[ESP_MAX_DNS];
```

Comment:
Allocating this is not really needed if `CONFIG_WIFI_ESP_DNS_USE` is not set, in which case we could save some bytes. Of course then you would need to add more `#ifdef` to the code.

Reply:
`ESP_MAX_DNS` is 0 when `CONFIG_WIFI_ESP_DNS_USE=n`. So it should not take any space.

Reply:
Indeed, missed that.